bin/tictactoe

```diff
@@ -1,3 +1,38 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+puts "Hello there! Welcome to Tic Tac Toe."
+
+go_again = true
+while go_again
+  puts "How many players? Enter 0, 1, or 2"
+  num_players = gets.chomp.to_i
+  "You're playing with #{num_players} players! Let's get started."
+  case num_players 
+  when 0
+    player_1 = Computer.new("X")
+    player_2 = Computer.new("O")
+  when 1
+    if rand(0...1)
+      puts "You get to go first."
+      player_1 = Human.new("X")
+      player_2 = Computer.new("O")
+    else
+      puts "The computer gets to go first."
+      player_2 = Human.new("O")
+      player_1 = Computer.new("X")
+    end
+  when 2 
+    player_1 = Human.new("X")
+    player_2 = Human.new("O")
+  end
+
+  Game.new(player_1, player_2).play
+
+  puts "Would you like to play again? Y/N"
+  if gets.chomp.upcase != "Y"
+    go_again = false
+    puts "Thanks for playing!"
+  end
+end
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.